src/crypto/crypto_util.cc

```diff
@@ -339,16 +340,27 @@ ByteSource& ByteSource::operator=(ByteSource&& other) noexcept {
   return *this;
 }
 
-std::unique_ptr<BackingStore> ByteSource::ReleaseToBackingStore() {
+std::unique_ptr<BackingStore> ByteSource::ReleaseToBackingStore(
+    Environment* env) {
   // It's ok for allocated_data_ to be nullptr but
   // only if size_ is zero.
   CHECK_IMPLIES(size_ > 0, allocated_data_ != nullptr);
+#if defined(V8_ENABLE_SANDBOX)
+  // If the v8 sandbox is enabled, then all array buffers must be allocated
+  // via the isolate. External buffers are not allowed. So, instead of wrapping
+  // the allocated data we'll copy it instead.
+  std::unique_ptr<BackingStore> ptr = ArrayBuffer::NewBackingStore(
+      env->isolate(), size(), BackingStoreInitializationMode::kUninitialized);
+  memcpy(ptr->Data(), allocated_data_, size());
+  OPENSSL_clear_free(allocated_data_, size_);
+#else
   std::unique_ptr<BackingStore> ptr = ArrayBuffer::NewBackingStore(
       allocated_data_,
       size(),
       [](void* data, size_t length, void* deleter_data) {
         OPENSSL_clear_free(deleter_data, length);
       }, allocated_data_);
+#endif
```

Comment:
This looks like it would be a good idea to add a helper to e.g. util.h that switches between the two variants in a single location rather than having `#ifdef`s for every call, no?

Reply:
Likely yes. I'm wanting to minimize the individual changes in this PR so we can make sure they are correct, but then we can likely extract out some utilities (maybe in `node_buffer.h`) that handle this in a more general way separately. That can be done separately I think?

Reply:
I added a TODO